File: frontend/src/shared/api/support.ts
const API =
  "https://api.kalitka.jesarion.com/api/v1";

export interface SupportMessage {
  id: string;
  sender: "client" | "operator" | "support";
  text: string;
  messageType: "text" | "image";
  imagePath: string | null;
  imageUrl: string | null;
  status: "sent" | "read";
  createdAt: string;
}

export async function getMessages() {
  const response = await fetch(
    `${API}/support`,
    {
      credentials: "include",
    },
  );

  if (!response.ok) {
    throw new Error(
      "Не удалось загрузить сообщения",
    );
  }

  return response.json() as Promise<
    SupportMessage[]
  >;
}

File: backend/app/schemas/support.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class SupportMessageCreate(BaseModel):
    text: str


class SupportMessageRead(BaseModel):
    id: UUID

    sender: str

    text: str

    message_type: str = "text"

    image_path: str | None = None

    status: str = "sent"

    created_at: datetime

    class Config:
        from_attributes = True
